### DISCUSSION TITLE
scikit-build Community Meeting #18, 2024-05-17

### DISCUSSION THREAD

**Discussion by thewtex** [scikit-build/scikit-build#1085] [Community meeting notes]
## Attending

- Matt McCormick
- Henry S
- Jean-Christophe Fillion-Robin
- Ryan Mast

## Discussion

- Ryan's using scikit-build for software for energy simulations. See https://docs.helics.org/en/latest/ 
- Henry is at PyCon
    - scikit-build in talk
        - Slides on [iscipynumpy.dev](https://iscinumpy.dev/)
        - Talk is recorded and will be online later
    - May attend Packaging Summit
    - Pinging for feedback on `dynamic_metadata`
- scikit-build talk accepted for SciPy Conf, proceeding submission in progress
- cibuildwheel
    - Python 3.13 shipped
    - uv support being considered
- opt-in to native build, flag `wheel.cmake` to `false`, `wheel.cmake`, build binary portion, pylib. See https://scikit-build-core.readthedocs.io/en/latest/configuration.html
    - pure + compiled example: `black`, `mypy`
    - pip uses `--no-binary` to build from sdist
- scikit-build-core
    - build tool args, `-L` for `ninja`
    - Free threaded, Python 3.13, manylinux with `t`
        - build an extension that doesn't have GIL support builtin
        - the extension needs to just use the threads
        - Stable ABI does not work with Free Threaded
            - Stable ABI 4 may support
- mirrors-prettier
    - Archived in April 2024. https://github.com/pre-commit/mirrors-prettier
    - _prettier made some changes that breaks plugins entirely_. See [mirrors-prettier@f62a70a3a](https://github.com/pre-commit/mirrors-prettier/commit/f62a70a3a7114896b062de517d72829ea1c884b6)
    - Possible path forward would be to compile prettier to wasm and use it through [wasmtime](https://github.com/bytecodealliance/wasmtime)
- Binding python with web assembly
    - https://wasm.itk.org/en/latest/ 
    - Cross platform tooling for building https://github.com/dockcross/dockcross#readme

**Comment by henryiii** [comment]
Wow, didn't realize the mirror was archived. Without any notice on how to replace it. I don't use plugins most of the time. [6.5k](https://github.com/search?type=code&q=path%3A.pre-commit-config.yaml+pre-commit%2Fmirrors-prettier+NOT+is%3Afork) projects use this hook.
